Python/Product/Analysis/Infrastructure/EnumerableExtensions.cs

```diff
@@ -48,5 +48,11 @@ public static bool IsNullOrEmpty<T>(this IEnumerable<T> source)
         public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T> source) where T : class {
             return source.Where(NotNull);
         }
+
+        public static bool SetEquals<T>(this IEnumerable<T> source, IEnumerable<T> other) where T : class {
+            var set1 = new HashSet<T>(source);
+            var set2 = new HashSet<T>(other);
+            return set1.Count == set2.Count && set1.All(x => set2.Contains(x));
```

Comment:
What's wrong with the SetEquals method of HashSet?

Also, I predict it will take about 5 minutes before we want to pass a comparer into this method (if only for case-insensitive string equality). Worth adding now (with a default to `null`)?

Reply:
Done